src/main/kotlin/org/rust/ide/intentions/MatchToIfLetIntention.kt

```diff
@@ -19,38 +18,57 @@ class MatchToIfLetIntention : RsElementBaseIntentionAction<MatchToIfLetIntention
     data class Context(
         val match: RsMatchExpr,
         val matchTarget: RsExpr,
-        val nonVoidArm: RsMatchArm,
-        val pat: RsPat
+        val matchBody: RsMatchBody
     )
 
     override fun findApplicableContext(project: Project, editor: Editor, element: PsiElement): Context? {
         val matchExpr = element.ancestorStrict<RsMatchExpr>() ?: return null
         if (element != matchExpr.match) return null
         val matchTarget = matchExpr.expr ?: return null
         val matchBody = matchExpr.matchBody ?: return null
-        val matchArmList = matchBody.matchArmList
+        if (matchBody.matchArmList.isEmpty()) return null
+        if (matchBody.matchArmList.any { it.matchArmGuard != null || it.outerAttrList.isNotEmpty() }) return null
 
-        val nonVoidArm = matchArmList.singleOrNull { it.expr?.isVoid == false } ?: return null
-        if (nonVoidArm.matchArmGuard != null || nonVoidArm.outerAttrList.isNotEmpty()) return null
-        val pat = nonVoidArm.pat
-
-        return Context(matchExpr, matchTarget, nonVoidArm, pat)
+        return Context(matchExpr, matchTarget, matchBody)
     }
 
     override fun invoke(project: Project, editor: Editor, ctx: Context) {
-        val (matchExpr, matchTarget, arm, pat) = ctx
+        val arms = ctx.matchBody.matchArmList
+        val lastArm = arms.lastOrNull() ?: return
+        val lastArmPatWild = lastArm.pat is RsPatWild
+        val lastArmExprEmpty = lastArm.expr?.let {
+            when {
+                it is RsUnitExpr -> true
+                it is RsBlockExpr && it.block.children.isEmpty() -> true
+                else -> false
+            }
+        } ?: false
+
+        val text = buildString {
+            for ((index, arm) in arms.withIndex()) {
+                var armText = "if let ${arm.pat.text} = ${ctx.matchTarget.text}"
```

Comment:
I think we should use `==` when possible, e.g `if x == 1` instead of `if let 1 = x`:

```rust
fn func(X: i32) {
    match x {
        1 => println!("1"),
        _ => println!("2"),
    };
}
```

Reply:
Would you mind if we leave this for a follow-up PR? It doesn't seem trivial and it's not strictly related to the generalization done by this PR.

Reply:
Ok, sure